src/components/Table/TableBody/TableBodyRow/TableBodyRow.jsx

```diff
@@ -208,51 +284,70 @@ const TableBodyRow = ({
       https://github.com/IBM/carbon-components-react/issues/1247
       Also move onClick logic above into TableSelectRow
       */}
-      <Checkbox id={`select-row-${id}`} labelText={selectRowText} hideLabel checked={isSelected} />
+      <span style={{ position: 'relative', left: nestingOffset }}>
+        <Checkbox
+          id={`select-row-${id}`}
+          labelText={selectRowText}
+          hideLabel
+          checked={isSelected}
+        />
+      </span>
     </StyledCheckboxTableCell>
   ) : null;
 
+  const firstVisibleColIndex = ordering.findIndex(col => !col.isHidden);
   const tableCells = (
     <React.Fragment>
       {rowSelectionCell}
-      {ordering.map(col => {
+      {ordering.map((col, idx) => {
         const matchingColumnMeta = columns && columns.find(column => column.id === col.columnId);
+        const offset = firstVisibleColIndex === idx ? nestingOffset : 0;
         return !col.isHidden ? (
           <StyledTableCellRow
             key={col.columnId}
             data-column={col.columnId}
+            data-offset={offset}
+            style={{ paddingRight: offset }}
             width={matchingColumnMeta && matchingColumnMeta.width}>
-            {col.renderDataFunction ? (
-              col.renderDataFunction({
-                // Call the column renderer if it's provided
-                value: children[col.columnId],
-                columnId: col.columnId,
-                rowId: id,
-                row: children,
-              })
-            ) : (
-              <TableCellRenderer>{children[col.columnId]}</TableCellRenderer>
-            )}
+            <span style={{ position: 'relative', left: offset }}>
```

Comment:
this makes me a little nervous as downstream consumers won't be able to override it with css.  Couldn't we do this up in a styled component css in js class?